storage/client_raft_test.go

```diff
@@ -2238,25 +2239,22 @@ func TestRemoveRangeWithoutGC(t *testing.T) {
 	// can continue to use its last known replica ID.
 	mtc.stopStore(0)
 	mtc.restartStore(0)
-	// Turn off the GC queue to ensure that the replica is deleted at
-	// startup instead of by the scanner. This is not 100% guaranteed
-	// since the scanner could have already run at this point, but it
-	// should be enough to prevent us from accidentally relying on the
-	// scanner.
-	mtc.stores[0].SetReplicaGCQueueActive(false)
 
-	// The Replica object is not recreated.
-	if _, err := mtc.stores[0].GetReplica(rangeID); err == nil {
-		t.Fatalf("expected replica to be missing")
-	}
+	util.SucceedsSoon(t, func() error {
+		// The Replica object should be removed.
+		if _, err := mtc.stores[0].GetReplica(rangeID); err == nil {
```

Comment:
NYC but prefer `!testutils.IsError` or direct error comparison to this construction.